### PR TITLE
Fix Windows import

### LINKED ISSUE #9436: Importing with "add to library..." not working on 3.6.0 on Windows 10 under certain conditions
Hello.

I've been experiencing a problem with the new import module under Windows.
I've been using darktable for a few weeks now. My usual workflow is copying my images from the SD card under a folder I create manually. Then I import my pictures in several programs (C1, Lightroom) and now darktable.
As I don't want my pictures to be moved, I use the "add to library..." function. This method worked perfectly in 3.4 both under Windows 10 and Ubuntu. I use a dual boot system, with a shared data 2TB SSD with my pictures.

Since updating darktable to 3.6.0, "add to library..." does not work under Windows 10, but it does under Ubuntu.
Darktable reports that a number of pictures have been imported, but they don't appear anywhere. In Ubuntu, the pictures just appear as expected.

After some testing, I've discovered that **the issue is caused by the use of accents in the path** . I'm spanish and my folders contain accents in some cases. For instance, if the pictures are in "d:\javimm\Downloads\Cámaras" , the pictures are not imported, but if I use a folder with no accents ie "d:\javimm\Downloads\Camaras", the pictures are then imported and eberything is fine, so the issue is caused by accents in the path.

Regards.

### LINKED ISSUE #9475: I can't Import any new collection on Win10
Hi, friends, 

Congratulations for this new step for this great plataforme of photo edition.

**Describe the bug/issue**
I've tried to install the last version 3.6.0.1 of Darktable and the main issue is that I CAN'T ADD any new collection. I can work with the existing collections but it's impossible to add a new one.

I follow the normal steps: select a folder, select some files, press the button, etc.  and then, it seems that Darktable is working (it even appears a messages saying that the process has finished correctly) but the new Collection never arrives and Darktable doesn't create any new .xmp files.

I've tried this process several times with two different installations: 
   * a FRESH installation of 3.6.0.1 with a new database and 
   * an UPDATE installation from 3.4.1 with the existing database

In both cases Darktable failed and it's impossible to add a new collection from my hard disk to the Darktable database.

**Platform**
Windows 10, 64-bits

I worked with darktable the 3.4.1
I've been trying to uptate to the new 3.6.0.1
I've been trying to install a fresh 3.6.0.1

Sincererly, 

Joan Miquel Viadé

### PR DESCRIPTION
**Pull request by wpferguson**
Changed file check to use utf16 pathname instead of utf8.  Fixes #9475. Fixes #9436

### CODE REVIEW

**Comment by Nilvus** [comment]
@wpferguson: Could you add "fixes #9436" too in your first post? Thanks

**Comment by wpferguson** [comment]
@ralfbrown last commit was for you, to save you one :)

**Comment by ralfbrown** [comment]
@wpferguson There's plenty to go around....  Unfortunately.

File: src/common/utility.c [hunk 1]
```diff
@@ -283,7 +283,15 @@ gboolean dt_util_test_image_file(const char *filename)
...
-  if(_stati64(filename, &stats)) return FALSE;
+
+  // the code this replaced used utf8 paths with no problem
+  // utf8 paths will not work in this context for no reason 
+  // that I can figure out, but converting utf8 to utf16 works
+  // fine.
+
+  wchar_t *wfilename = g_utf8_to_utf16(filename, -1, NULL, NULL, NULL);
+  if(_wstati64(wfilename, &stats)) return FALSE;
+  g_free(wfilename);
```
Comment (TurboGit) [inline src/common/utility.c:294]: you will leak memory if return FALSE above.

**Review by TurboGit — CHANGES_REQUESTED** [review]
memory leak to fix.

**Comment by Nilvus** [comment]
@wpferguson: you should add "fixes" before #9436 to link the issue. It's not linked without.

**Comment by jenshannoschwalm** [comment]
@wpferguson i looked up some docs about the problem and indeed your finding is not something weird but it seems this has been a true bug before. See:

https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions?view=msvc-160

Under remarks you find that the function expects a wide-character string.

File: src/common/utility.c (hunk 1 above)
Comment (wpferguson) [reply src/common/utility.c:294]: Oops!  Fixed :)

File: src/common/utility.c [hunk 2]
```diff
@@ -283,8 +283,17 @@ gboolean dt_util_test_image_file(const char *filename)
...
-  if(_stati64(filename, &stats)) return FALSE;
-#else
+
+  // the code this replaced used utf8 paths with no problem
+  // utf8 paths will not work in this context for no reason 
+  // that I can figure out, but converting utf8 to utf16 works
+  // fine.
+
+  wchar_t *wfilename = g_utf8_to_utf16(filename, -1, NULL, NULL, NULL);
+  int result = _wstati64(wfilename, &stats);
```
Comment (TurboGit) [inline src/common/utility.c:293]: a little const here and we are set !

Comment (wpferguson) [reply src/common/utility.c]: Done, and thanks

**Review by TurboGit — APPROVED** [review]
Perfect, thanks for handling this Win32 specific issue!

**Pull request merged**